superset/databases/filters.py

```diff
@@ -27,19 +27,30 @@ class DatabaseFilter(BaseFilter):
     # TODO(bogdan): consider caching.
     def schema_access_databases(self) -> Set[str]:  # noqa pylint: disable=no-self-use
         return {
-            security_manager.unpack_schema_perm(vm)[0]
+            security_manager.unpack_perm(vm)[0]
             for vm in security_manager.user_view_menu_names("schema_access")
         }
 
+    def datasource_access_databases(  # noqa pylint: disable=no-self-use
+        self,
+    ) -> Set[str]:
+        return {
+            security_manager.unpack_perm(vm)[0]
```

Comment:
nit: not something you introduced, but while we're touching up the code and for readability's sake, it would be nice to use a `NamedTuple` or `@dataclass` as the return type in `unpack_perm` so we can replace `[0]` with `.database_name` here and `[1]` with `.schema_name` when requesting the schema name.